### ISSUE TITLE
health: remove all per-plugin health status

### ISSUE THREAD

**Issue by miekg** [coredns/coredns#2534]
Using health to convey per plugin health status might make sense, but it get's tricky fast. Anything that requires network to be up, will in the event of a network hiccup report unhealthly. As this is a per process, the whole of coredns goes unhhealthy.

I..e. k8s can't connect to api -> unhealthy, while the cache might be perfectly fine to hand out requests in the mean time. 

Also when we report health OK; we're ready to process queries; if some plugin is not fully ready we return SERVFAIL, perfectly acceptable way of doing that in the DNS and the caller should know how to deal with that; retry; other server; retry again; etc.

IOW: per plugin health is useless and should be removed; all health code and be removed and concentrated in the (original) health plugin that says: We're up and running; some plugins might not be fully ready yet, but those will return SERVFAIL.

**Comment by chrisohaver** [comment]
For k8s, would you suggest a new plugin to report readiness?
Or have the k8s plugin open a port and report it directly?

**Comment by miekg** [comment]
Moving this to readiness might make a lot of sense. Have to figure out of health add anything at all, and what happens in k8s when readiness is toggling between on and off

**Comment by miekg** [comment]
From k8s docs:

> Sometimes, applications are temporarily unable to serve traffic. For example, an application might need to load large data or configuration files during startup, or depend on external services after startup. In such cases, you don’t want to kill the application, but you don’t want to send it requests either. Kubernetes provides readiness probes to detect and mitigate these situations. A pod with containers reporting that they are not ready does not receive traffic through Kubernetes Services.

TL;DR: just a bad, cache might be perfectly fine in handling this, but we nack health for reasons and don't get any queries. That just leads to sadness.
